### PR TITLE
PAL-411 attribute-masking-service integration tests

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: attribute-masking-service-integration/pom.xml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2019 Crown Copyright
```
Comment (pd104923) [inline attribute-masking-service-integration/pom.xml:3]:
```suggestion
  ~ Copyright 2020 Crown Copyright
```

File: attribute-masking-service-integration/src/test/resources/application.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+spring:
+  application:
+    name: audit-service
```
Comment (pd104923) [inline attribute-masking-service-integration/src/test/resources/application.yaml:17]:
```suggestion
    name: attribute-masking-service
```

File: attribute-masking-service-integration/src/test/resources/application.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  output:
+    ansi:
+      enabled: always
+  main:
+    allow-bean-definition-overriding: true
+
+server:
+  port: 8088
```
Comment (pd104923) [inline attribute-masking-service-integration/src/test/resources/application.yaml:25]: Have we decided if there is going to be a specific port numbers for these services? This one is the same as the audit-service so would need to be changed.

**Review by pd104923 — CHANGES_REQUESTED** [review]

**Review by pd104923 — APPROVED** [review]

File: attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import uk.gov.gchq.palisade.service.attributemask.request.AttributeMaskingResponse;
+import uk.gov.gchq.palisade.service.attributemask.request.StreamMarker;
+import uk.gov.gchq.palisade.service.attributemask.request.Token;
+import uk.gov.gchq.palisade.service.attributemask.service.AttributeMaskingService;
+import uk.gov.gchq.palisade.service.attributemask.web.AttributeMaskingController;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(classes = AttributeMaskingApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
+class AttributeMaskingServiceComponentTest {
```
Comment (m78233) [inline attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java:39]: Suggest making class public.

File: attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    @Autowired
+    private TestRestTemplate restTemplate;
```
Comment (m78233) [inline attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java:42]: Think that there still needs to be a test for the Controller.  Managed to find one ok (ish) article on the distinction of using TestRestTemplate and MockMvc in testing.  
https://spring.io/blog/2012/11/12/spring-framework-3-2-rc1-spring-mvc-test-framework.  Basic is that the MockMvc is for the server-side and TestRestTemplate is for the client-side.

File: attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java (hunk 4 above)
Comment (m78233) [inline attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java:39]: We have to create a class in one repo and related Component test in another repo.  It is a scenario that can easily create problems if the two get out of sync.  Think that it should in the same repo as the code that is being tested.

**Review by m78233 — COMMENTED** [review]
Suggestion, change ticket for just the  Rest endpoint.  Ticket refers to Kafka Consumer and Producer and Contract test be put on another ticket.

File: attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java (hunk 4 above)
Comment (dev930018) [reply attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java:39]: Junit 5 and sonarqube take the stance that test classes should be package-private

Comment (dev930018) [reply attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java:39]: Will be covered in a future ticket

File: attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java (hunk 5 above)
Comment (dev930018) [reply attribute-masking-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/attributemask/AttributeMaskingServiceComponentTest.java:42]: I feel in this case there isn't much to gain from these sorts of tests - the application functionality just isn't in here yet, and it forces us back to JUnit 4 methods afaik

**Pull request closed without merging**